### ISSUE TITLE
Keyboard navigation in Spreadsheet is broken when the display scale is different than 100%

### ISSUE THREAD

**Issue by nelito987** [telerik/kendo-ui-core#7541]
### Bug report

When the scale and layout setting of the machine is different than 100% the keyboard navigation does not work properly. The issue occurs with different scales on each machine. For example, on some machines, it could work with all scales, but on others, it may not work with a scale of 125% or 175%, even if this is the recommended value. 

### Reproduction of the problem
1. Open the demos - https://demos.telerik.com/kendo-ui/spreadsheet/keyboard-navigation
2. Set a scale different than 100% to the computer
![image](https://github.com/telerik/kendo-ui-core/assets/14364791/efb2c6a3-2c8c-4051-ab4c-a7a084ed2e39)

Note, that you may need to test with different scales to manage to replicate the issue.

3. Using the arrow keys navigate to the last column of the Spreadsheet (AX)
4. Go to the previous column (AW)
5. Try to move up/down/left/right

### Current behavior
The navigation does not work. The cell selection does not work or goes to a wrong cell.

https://somup.com/c06nQJBFmV

### Expected/desired behavior
The Spreadsheet navigation should work with all screen scales.

### Environment

* **Kendo UI version:** 2023.2.829
* **Browser:** [all ]

**Comment by nelito987** [comment]
Reported also in ticket ID: 1683544
